jaydebeapi/__init__.py

```diff
@@ -486,9 +487,12 @@ def _close_last(self):
 
     def _set_stmt_parms(self, prep_stmt, parameters):
         for i in range(len(parameters)):
-            # print (i, parameters[i], type(parameters[i]))
             prep_stmt.setObject(i + 1, parameters[i])
 
+        # https://docs.oracle.com/javase/7/docs/api/java/sql/Statement.html#setQueryTimeout(int)
+        if self._timeout is not None:
+            prep_stmt.setQueryTimeout(int(self._timeout))  # unit: seconds
```

Comment:
I would prefer to have it fail fast: Put the conversion into the connect method and only pass down the integer so you don't need a conversion down here.